tools/smithy-rs-sync/src/main.rs

```diff
@@ -0,0 +1,409 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+use anyhow::{anyhow, bail, Context, Result};
+use git2::{Commit, IndexAddOption, ObjectType, Oid, Repository, ResetType, Signature};
+use std::fs::OpenOptions;
+use std::io::{Read, Write};
+use std::path::{Path, PathBuf};
+use std::process::Command;
+use std::time::Instant;
+use structopt::StructOpt;
+
+#[derive(StructOpt, Debug)]
+#[structopt(name = "smithy-rs-sync")]
+/// A CLI tool to replay commits from smithy-rs, generate code, and commit that code to aws-rust-sdk
+struct Opt {
+    /// The path to the smithy-rs repo folder
+    #[structopt(long, parse(from_os_str))]
+    smithy_rs: PathBuf,
+    /// The path to the aws-sdk-rust folder
+    #[structopt(long, parse(from_os_str))]
+    aws_sdk: PathBuf,
+    /// The branch in aws-sdk-rust that commits will be mirrored to
+    #[structopt(long, default_value = "next")]
+    branch: String,
+}
+
+const BOT_NAME: &str = "AWS SDK Rust Bot";
+const BOT_EMAIL: &str = "aws-sdk-rust-primary@amazon.com";
+const COMMIT_HASH_FILENAME: &str = ".smithyrs-githash";
+
+/// A macro for attaching info to error messages pointing to the line of code responsible for the error
+/// [Thanks to dtolnay for this macro](https://github.com/dtolnay/anyhow/issues/22#issuecomment-542309452)
+macro_rules! here {
+    () => {
+        concat!("error at ", file!(), ":", line!(), ":", column!())
+    };
+}
+
+/// Run this app in order to keep aws-sdk-rust in sync with smithy-rs.
+///
+/// pre-requisites:
+/// - an up-to-date local copy of smithy-rs repo
+/// - an up-to-date local copy of aws-sdk-rs repo
+/// - a Unix-ey system (for the `cp` and `rf` commands to work)
+/// - Java Runtime Environment v11 (in order to run gradle commands)
+///
+/// ```sh
+/// cargo run -- \
```

Comment:
```suggestion
/// cargo run \
```

Reply:
you need the `--` so that the arguments go to the binary and not to cargo

Reply:
I need that there so that cargo doesn't eat up the args. The `--` makes them get interpreted by the program being run instead.